### PR TITLE
New motion controller implementation

### PR DESCRIPTION
**Pull request by superpingu**
Contains a rewrite, almost from scratch, of the motion algorithm. Unlike the previous version, it actually works.

### CODE REVIEW

File: robot/Makefile [hunk 1]
```diff
@@ -122,9 +123,10 @@ CSRC = $(STARTUPSRC) \
        $(BOARDSRC) \
        $(RADIOSRC) \
        $(DRIVERSRC) \
+       $(RTTSRC) \
        $(IMU) \
-       $(MOVING) \
-       trigo.c led.c main.c
+	   $(MOTION) \
```
Comment (samueltardieu) [inline robot/Makefile]: Bad indentation.

File: robot/main.c [hunk 2]
```diff
@@ -1,41 +1,34 @@
...
-    start_moving();
-	startFusion();
+	initSequencer();
+	initMotion();
 
-	while(1) {
-		chThdSleepMilliseconds(5000);
-	}
+	while(1)
+		chThdSleepMilliseconds(1000);
```
Comment (samueltardieu) [inline robot/main.c]: Why wake up the thread every second instead of sleeping indefinitely?

File: robot/motion.c [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+	point_t dest = {destMove->x, destMove->y};
+
+	// both circles cannot be of the same radius because of the way we compute tangents
+	if(startRadius == endRadius)
+		startRadius++;
+
+	computeCenters(&dep, depDir, startRadius, &depCenters[0], &depCenters[1]);
+	computeCenters(&dest, destDir, endRadius, &destCenters[0], &destCenters[1]);
+
+	for(i=0; i<2; i++) { // try the 2 possible departure circles
```
Comment (samueltardieu) [inline robot/motion.c]: Loop variables should be declared when used, as in
``` c
    for (int i = 0; i < 2; i++)
```

We are in 2017, and this has been introduced in C99.

**Review by samueltardieu — CHANGES_REQUESTED** [review]

**Comment by samueltardieu** [comment]
Please fix the appropriate commits (using `git rebase -i`) and then `push -f` to update this PR.

Also, in the commit which remove the duplicate `sinTab`, don't forget to include a `fix #39` message to auto-close the issue.

**Comment by superpingu** [comment]
Done.

**Pull request merged**